test/Nat/log.lean

```diff
@@ -0,0 +1,3 @@
+import Mathlib.Data.Nat.Log
+
+#eval Nat.log 2 (2 ^ 10000000)
```

Comment:
We should document the reason for passing of this test now with a reference to `log_eq_logC`.

Also, the test folder got moved into a separate package `./MathlibTest`. You can just copy the file there.